In [1]:
#load date
from sklearn.datasets import load_iris
iris_datasets = load_iris()

In [2]:
print("Jenis datasets iris = \n {}".format(iris_datasets.keys()))

Jenis datasets iris = 
 dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])


In [3]:
print("Target Name = {}".format(iris_datasets['target_names']))

Target Name = ['setosa' 'versicolor' 'virginica']


In [4]:
print("Target name = {}".format(iris_datasets['target']))
len(iris_datasets['target']) #menghitung jumblah array

Target name = [0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
 0 0 0 0 0 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1
 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2]


150

In [5]:
print("target Name = {}".format(iris_datasets['feature_names']))

target Name = ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']


In [6]:
print("Data = {}".format(iris_datasets['data']))
print('ukuran data = {}'.format(iris_datasets['data'].shape))

Data = [[5.1 3.5 1.4 0.2]
 [4.9 3.  1.4 0.2]
 [4.7 3.2 1.3 0.2]
 [4.6 3.1 1.5 0.2]
 [5.  3.6 1.4 0.2]
 [5.4 3.9 1.7 0.4]
 [4.6 3.4 1.4 0.3]
 [5.  3.4 1.5 0.2]
 [4.4 2.9 1.4 0.2]
 [4.9 3.1 1.5 0.1]
 [5.4 3.7 1.5 0.2]
 [4.8 3.4 1.6 0.2]
 [4.8 3.  1.4 0.1]
 [4.3 3.  1.1 0.1]
 [5.8 4.  1.2 0.2]
 [5.7 4.4 1.5 0.4]
 [5.4 3.9 1.3 0.4]
 [5.1 3.5 1.4 0.3]
 [5.7 3.8 1.7 0.3]
 [5.1 3.8 1.5 0.3]
 [5.4 3.4 1.7 0.2]
 [5.1 3.7 1.5 0.4]
 [4.6 3.6 1.  0.2]
 [5.1 3.3 1.7 0.5]
 [4.8 3.4 1.9 0.2]
 [5.  3.  1.6 0.2]
 [5.  3.4 1.6 0.4]
 [5.2 3.5 1.5 0.2]
 [5.2 3.4 1.4 0.2]
 [4.7 3.2 1.6 0.2]
 [4.8 3.1 1.6 0.2]
 [5.4 3.4 1.5 0.4]
 [5.2 4.1 1.5 0.1]
 [5.5 4.2 1.4 0.2]
 [4.9 3.1 1.5 0.2]
 [5.  3.2 1.2 0.2]
 [5.5 3.5 1.3 0.2]
 [4.9 3.6 1.4 0.1]
 [4.4 3.  1.3 0.2]
 [5.1 3.4 1.5 0.2]
 [5.  3.5 1.3 0.3]
 [4.5 2.3 1.3 0.3]
 [4.4 3.2 1.3 0.2]
 [5.  3.5 1.6 0.6]
 [5.1 3.8 1.9 0.4]
 [4.8 3.  1.4 0.3]
 [5.1 3.8 1.6 0.2]
 [4.6 3.2 1.4 0.2]
 [5.3 3.7 1.5 0.2]
 [5.  3.3 1.4 0.2]
 [7.  3.2 4.7 1.4]
 [6.4 3.2 4.5 1.5]
 [6.9

PROSES TRAINING DAN TESTING
membagi 159 data untuk di bagi menjadi data trainning (85%) dan testing (15%)


In [8]:
from sklearn.model_selection import train_test_split
data = iris_datasets['data'] #x
target = iris_datasets['target'] #y

x_train, x_test, y_train, y_test = train_test_split(data, target, test_size=0.15, random_state=0)

In [9]:
print("jumlah data x training =",x_train.shape)
print("jumlah data x testing =",x_test.shape)
print("jumlah data y training =",y_train.shape)
print("jumlah data y testing =",y_test.shape)

jumlah data x training = (127, 4)
jumlah data x testing = (23, 4)
jumlah data y training = (127,)
jumlah data y testing = (23,)


In [10]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=1)

In [11]:
knn.fit(x_train,y_train) #knn

KNeighborsClassifier(n_neighbors=1)

In [12]:
import numpy as np
x_baru = np.array([[3, 5.1, 5, 4.2]])
prediksi = knn.predict(x_baru)
print("Hasil Prediksi KNN = ",prediksi)

Hasil Prediksi KNN =  [2]


In [13]:
x_test

array([[5.8, 2.8, 5.1, 2.4],
       [6. , 2.2, 4. , 1. ],
       [5.5, 4.2, 1.4, 0.2],
       [7.3, 2.9, 6.3, 1.8],
       [5. , 3.4, 1.5, 0.2],
       [6.3, 3.3, 6. , 2.5],
       [5. , 3.5, 1.3, 0.3],
       [6.7, 3.1, 4.7, 1.5],
       [6.8, 2.8, 4.8, 1.4],
       [6.1, 2.8, 4. , 1.3],
       [6.1, 2.6, 5.6, 1.4],
       [6.4, 3.2, 4.5, 1.5],
       [6.1, 2.8, 4.7, 1.2],
       [6.5, 2.8, 4.6, 1.5],
       [6.1, 2.9, 4.7, 1.4],
       [4.9, 3.6, 1.4, 0.1],
       [6. , 2.9, 4.5, 1.5],
       [5.5, 2.6, 4.4, 1.2],
       [4.8, 3. , 1.4, 0.3],
       [5.4, 3.9, 1.3, 0.4],
       [5.6, 2.8, 4.9, 2. ],
       [5.6, 3. , 4.5, 1.5],
       [4.8, 3.4, 1.9, 0.2]])

In [14]:
prediksi1 = knn.predict(x_test)
print("Hasil prediksi x_test =  ",prediksi1)

Hasil prediksi x_test =   [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0]


In [15]:
y_test

array([2, 1, 0, 2, 0, 2, 0, 1, 1, 1, 2, 1, 1, 1, 1, 0, 1, 1, 0, 0, 2, 1,
       0])

EVALUASI METODE MECHINE LEARNING

In [16]:
from sklearn.metrics import accuracy_score
print('Hasil Akurasi = ',accuracy_score(prediksi1,y_test)*100)

Hasil Akurasi =  100.0


PROSES TREANING DAN TESTING
membagi 150 data untuk di bagi menjadi data traning(74,5%) dan testing(25,5%)

In [17]:
from sklearn.model_selection import train_test_split
data = iris_datasets['data'] #x
target = iris_datasets['target'] #y

x_train, x_test, y_train, y_test = train_test_split(data, target, test_size=0.255, random_state=0)

In [18]:
print("jumlah data x training =",x_train.shape)
print("jumlah data x testing =",x_test.shape)
print("jumlah data y training =",y_train.shape)
print("jumlah data y testing =",y_test.shape)

jumlah data x training = (111, 4)
jumlah data x testing = (39, 4)
jumlah data y training = (111,)
jumlah data y testing = (39,)


PROSES TRANING

In [19]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=1)

In [20]:
knn.fit(x_train,y_train) #knn

KNeighborsClassifier(n_neighbors=1)

MELAKUKAN PREDIKSI

In [21]:
import numpy as np
x_baru = np.array([[3, 5.1, 5, 4.2]])
prediksi = knn.predict(x_baru)
print("Hasil Prediksi KNN = ",prediksi)

Hasil Prediksi KNN =  [2]


In [22]:
x_test

array([[5.8, 2.8, 5.1, 2.4],
       [6. , 2.2, 4. , 1. ],
       [5.5, 4.2, 1.4, 0.2],
       [7.3, 2.9, 6.3, 1.8],
       [5. , 3.4, 1.5, 0.2],
       [6.3, 3.3, 6. , 2.5],
       [5. , 3.5, 1.3, 0.3],
       [6.7, 3.1, 4.7, 1.5],
       [6.8, 2.8, 4.8, 1.4],
       [6.1, 2.8, 4. , 1.3],
       [6.1, 2.6, 5.6, 1.4],
       [6.4, 3.2, 4.5, 1.5],
       [6.1, 2.8, 4.7, 1.2],
       [6.5, 2.8, 4.6, 1.5],
       [6.1, 2.9, 4.7, 1.4],
       [4.9, 3.6, 1.4, 0.1],
       [6. , 2.9, 4.5, 1.5],
       [5.5, 2.6, 4.4, 1.2],
       [4.8, 3. , 1.4, 0.3],
       [5.4, 3.9, 1.3, 0.4],
       [5.6, 2.8, 4.9, 2. ],
       [5.6, 3. , 4.5, 1.5],
       [4.8, 3.4, 1.9, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [6.2, 2.8, 4.8, 1.8],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.8, 1.9, 0.4],
       [6.2, 2.9, 4.3, 1.3],
       [5. , 2.3, 3.3, 1. ],
       [5. , 3.4, 1.6, 0.4],
       [6.4, 3.1, 5.5, 1.8],
       [5.4, 3. , 4.5, 1.5],
       [5.2, 3.5, 1.5, 0.2],
       [6.1, 3. , 4.9, 1.8],
       [6.4, 2

In [23]:
prediksi1 = knn.predict(x_test)
print("Hasil prediksi x_test =  ",prediksi1)

Hasil prediksi x_test =   [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2 1]


In [24]:
y_test

array([2, 1, 0, 2, 0, 2, 0, 1, 1, 1, 2, 1, 1, 1, 1, 0, 1, 1, 0, 0, 2, 1,
       0, 0, 2, 0, 0, 1, 1, 0, 2, 1, 0, 2, 2, 1, 0, 1, 1])

EVALUASI MECHINE LEARNING

In [25]:
from sklearn.metrics import accuracy_score
print('Hasil Akurasi = ',accuracy_score(prediksi1, y_test)*100)

Hasil Akurasi =  97.43589743589743


PROSES TRENING DAN TESTING
membagi 150 data untuk di bagi menjadi data traning (68,75%) dan testing ( 31,25%)

In [26]:
from sklearn.model_selection import train_test_split
data = iris_datasets['data'] #x
target = iris_datasets['target'] #y

x_train, x_test, y_train, y_test = train_test_split(data, target, test_size=0.3125, random_state=0)

In [27]:
print("jumlah data x training =",x_train.shape)
print("jumlah data x testing =",x_test.shape)
print("jumlah data y training =",y_train.shape)
print("jumlah data y testing =",y_test.shape)

jumlah data x training = (103, 4)
jumlah data x testing = (47, 4)
jumlah data y training = (103,)
jumlah data y testing = (47,)


PROSES TRENING

In [28]:
from sklearn.neighbors import KNeighborsClassifier
knn = KNeighborsClassifier(n_neighbors=1)

In [29]:
knn.fit(x_train,y_train) #knn

KNeighborsClassifier(n_neighbors=1)

MELAKUKAN PREDIKSI

In [30]:
import numpy as np
x_baru = np.array([[3, 5.1, 5, 4.2]])
prediksi = knn.predict(x_baru)
print("Hasil Prediksi KNN = ",prediksi)

Hasil Prediksi KNN =  [2]


In [31]:
x_test

array([[5.8, 2.8, 5.1, 2.4],
       [6. , 2.2, 4. , 1. ],
       [5.5, 4.2, 1.4, 0.2],
       [7.3, 2.9, 6.3, 1.8],
       [5. , 3.4, 1.5, 0.2],
       [6.3, 3.3, 6. , 2.5],
       [5. , 3.5, 1.3, 0.3],
       [6.7, 3.1, 4.7, 1.5],
       [6.8, 2.8, 4.8, 1.4],
       [6.1, 2.8, 4. , 1.3],
       [6.1, 2.6, 5.6, 1.4],
       [6.4, 3.2, 4.5, 1.5],
       [6.1, 2.8, 4.7, 1.2],
       [6.5, 2.8, 4.6, 1.5],
       [6.1, 2.9, 4.7, 1.4],
       [4.9, 3.6, 1.4, 0.1],
       [6. , 2.9, 4.5, 1.5],
       [5.5, 2.6, 4.4, 1.2],
       [4.8, 3. , 1.4, 0.3],
       [5.4, 3.9, 1.3, 0.4],
       [5.6, 2.8, 4.9, 2. ],
       [5.6, 3. , 4.5, 1.5],
       [4.8, 3.4, 1.9, 0.2],
       [4.4, 2.9, 1.4, 0.2],
       [6.2, 2.8, 4.8, 1.8],
       [4.6, 3.6, 1. , 0.2],
       [5.1, 3.8, 1.9, 0.4],
       [6.2, 2.9, 4.3, 1.3],
       [5. , 2.3, 3.3, 1. ],
       [5. , 3.4, 1.6, 0.4],
       [6.4, 3.1, 5.5, 1.8],
       [5.4, 3. , 4.5, 1.5],
       [5.2, 3.5, 1.5, 0.2],
       [6.1, 3. , 4.9, 1.8],
       [6.4, 2

In [32]:
prediksi1 = knn.predict(x_test)
print("Hasil prediksi x_test =  ",prediksi1)

Hasil prediksi x_test =   [2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 2 1 0 0 2 0 0 1 1 0 2 1 0 2 2 1 0
 2 1 1 2 0 2 0 0 1 2]


In [34]:
y_test

array([2, 1, 0, 2, 0, 2, 0, 1, 1, 1, 2, 1, 1, 1, 1, 0, 1, 1, 0, 0, 2, 1,
       0, 0, 2, 0, 0, 1, 1, 0, 2, 1, 0, 2, 2, 1, 0, 1, 1, 1, 2, 0, 2, 0,
       0, 1, 2])

EVALUASI METODE MECHINE LEARNIG

In [35]:
from sklearn.metrics import accuracy_score
print('Hasil Akurasi = ',accuracy_score(prediksi1, y_test)*100)

Hasil Akurasi =  97.87234042553192
